client.go

```diff
@@ -129,7 +132,7 @@ func NewCustomClient(appKey, host, scheme string) (*Client, error) {
 	case "pusher:connection_established":
 		sChannels := new(subscribedChannels)
 		sChannels.channels = make([]string, 0)
-		pClient := Client{ws, make(chan *Event, EVENT_CHANNEL_BUFF_SIZE), make(chan bool), sChannels, make(map[string]chan *Event)}
+		pClient := Client{ws, make(chan *Event, EVENT_CHANNEL_BUFF_SIZE), make(chan bool), sChannels, make(map[string]chan *Event), make(chan *error)}
```

Comment:
The chan should be buffered, so that writes don't block if the reader isn't actively trying to read from it.